tests/services/test_users.py

```diff
@@ -375,3 +498,101 @@ def test_update_user_when_get_unhandled_internal_server_error_raises_base_error(
         mock_connection.put.side_effect = internal_server_error
         with pytest.raises(Py42InternalServerError):
             user_service.update_user("123", username="foo")
+
+    @patch.object(
```

Comment:
Just found out about the `patch.object` decorator/method!  Very helpful for patching out a helper method that makes an api call